source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -981,19 +981,29 @@ OcspStapleAction ServerContextImpl::ocspStapleAction(const CertContext& cert_con
 
 int ServerContextImpl::handleOcspStapling(SSL* ssl, void*) {
   const bool client_ocsp_capable = isClientOcspCapable(ssl);
-  // returns server cert selected for this connection
-  // see https://www.openssl.org/docs/man1.1.1/man3/SSL_set_tlsext_status_ocsp_resp.html
-  // for details
-  auto* selected_cert = SSL_get_certificate(ssl);
-  const auto& cert_context = certificateContext(selected_cert);
-  auto ocsp_staple_action = ocspStapleAction(cert_context, client_ocsp_capable);
+  
+  // Fix OSSM-2208 by looping on all certificates to find at least a good one
+  const CertContext* selected_cert_context = &tls_context_.cert_contexts_[0];
```

Comment:
The `selected_cert_context` and `ocsp_staple_action` variables don't really need to be initialised before the loop. They will get initialised to those same values in the first iteration of the loop anyway.

Reply:
Ok, but I've taken it from the upstream Envoy 1.24

Reply:
/retest